### ISSUE TITLE
Ugly rendering of the output visualization of TeXworks

### ISSUE THREAD

**Issue by SSebastiano** [TeXworks/texworks#1037]
**Bug description:** bad rendering with recent version of TeXworks 0.8.3.

**Steps to reproduce the problem:**

1. Compile a .tex file using TikZ syntax with TeXworks 0.8.3.

**Expected behavior:**
A non clear output with the white vertical line, with the strange borders.

**General information:**
_TeXworks version_: 0.8.3.
_TeXworks obtained from_: MikTeX of 2024-01-27
_Operating system_: Windows 1o

**Additional information:** See the website at  https://tex.stackexchange.com/questions/707606/ugly-rendering-of-the-output-visualization-of-texworks

**Comment by stloeffler** [comment]
Thanks for reporting. I do have a few questions/comments:
1) There is no version 0.8.3 of TeXworks (yet). Do you mean 0.6.8?
2) TeXworks uses the poppler library for rendering, so it might depend on which poppler version was used for building
2) Does the problem persist when using the [latest development version](https://nightly.link/TeXworks/texworks/workflows/cd/main/TeXworks-Windows10)?

**Comment by SSebastiano** [comment]
Hi. I have written a wrong number:

1. the version is 0.6.8 of TeXworks (updated 2024-1-28 on my laptop). After I will change the number in the TeXworks issue on github. 

2. I don't know.

3. The problem persist from November-December 2023 (there was always the version 0.6.8 of TeXworks.

I hope the resolution of the problem of rendering of the output done with a LaTeX code.

My best regards.

Sebastiano

**Comment by stloeffler** [comment]
Thanks for clarifying. Still, could you please test the latest development build (as linked above, not distributed by MikTeX yet), which should show 0.7 as (preliminary) version number? It should be no problem having the two versions side by side (in fact, please don't overwrite the MikTeX TeXworks binary as that probably has some other MikTeX-specific modifications).

**Comment by matteofg** [comment]
It's possible to get the latest development version for macos?

**Comment by SSebastiano** [comment]
@stloeffler Dear Stefan Löffler I read your comment carefully. I have not tested version 0.7 at the [link](https://nightly.link/TeXworks/texworks/workflows/cd/main/TeXworks-Windows10) because I have feared to cause the damage in the distribution of MikTeX which works just fine and because I would not like to create two separate TeXworks between them. I, at this point, am waiting for the MikTeX developers to upgrade the version of TeXworks from 0.6.8 to 0.7. I wish you the best.
Regards and I hope as soon as possible that the problem will be solved. I await your news update.

**Comment by stloeffler** [comment]
@matteofg Yes, there's mac versions as well. Please have a look at <https://www.tug.org/texworks/#Online_resources>
@SSebastiano I fully understand. As I wrote, there _should_ be no harm in having two different versions (just don't overwrite your old one), but of course there's always a potential risk. The MikTeX devs likely won't adopt the current 0.7 development version, but rather the next stable release (which will be called 0.6.9 - and be based on the current 0.7 version) when it releases.

**Comment by SSebastiano** [comment]
@stloeffler Very kind Stefan Löffler we know that MiKTeX, is development by Christian Schenk. Who will notify him that the version that version 0.6.8 of TeXworks has problems and that version 0.6.9 needs to be released at the next update of MikTeX?

**Comment by SSebastiano** [comment]
@stloeffler Here there is my issue done on MikTeX related to TeXworks: https://github.com/MiKTeX/miktex/issues/1435

**Comment by stloeffler** [comment]
I will notify distributors (including MiKTeX devs) routinely whenever a new version of TeXworks is released.

From your links (the StackExchange post and the MiKTeX issue), it seems to me that this problem appeared recently. If so, this is likely not (directly) a bug in the TeXworks code, as the 0.6.8 release is about a year old by now and has not received any patches in that time.
As I speculated above, the problem may be related to an updated build with different (newer) libraries - in particular the poppler library used for rendering pdfs - but without knowing exactly which versions of which libraries are affected and which are not, it will be impossible to further analyze this issue.

**Comment by SSebastiano** [comment]
@stloeffler Thank you for your precious work and help for me. I will wait the last updated of MikTeX with new version of TeXworks. Best regards.

**Comment by SSebastiano** [comment]
@stloeffler Hi. Do you have news for new TeXworks. I have always the same bad rendering. My best regards.

**Comment by SSebastiano** [comment]
Hi. Do you have news for new TeXworks. I have always the same bad rendering using TeXworks. What I must doing? My best regards.

**Comment by stloeffler** [comment]
TeXworks 0.6.9 has recently been released and I have created MiKTeX/miktex#1450.
Whether this resolves your issue I don't know, however. IIRC, there were no significant changes to the rendering code from me, but as mentioned above, maybe some library updates will do the trick.

**Comment by SSebastiano** [comment]
@edocevoli @stloeffler The problem of rendering since 2024-3-19 yet is evident. What is the solution for 
Christian Schenk? I wait an answer. Bye. Sebastiano

**Comment by tbackdahl** [comment]
I had the same problem with a clean installation today of MiKTeX on windows 10. TexWorks version 0.6.9.
I got rid of the problem by manually replacing 
AppData\Local\Programs\MiKTeX\miktex\bin\x64\miktex-texworks.exe
with the TeXworks.exe downloaded from
https://nightly.link/TeXworks/texworks/workflows/cd/main/TeXworks-Windows10
The version is TeXworks-win-0.7.0-202408010314-git_840fd91
This is not a good way to install files in the MiKTeX system, but it helped me for now.
I had the same problem with a few older versions too, but I can not tell if it is the version number or who compiled it that made the difference.
The corresponding tickets on the MiKTeX github seems to be closed even though the problem persists.
Can someone with knowledge check if the MiKTeX team are using the same libraries as you?

**Comment by SSebastiano** [comment]
@tbackdahl Hi Thomas. Thank you very much for your response and for your help. Reading your post I inform you that I have the same problem after that I have written this post and written to maintainer of MikTeX and of TeXworks. In fact the version of TeXworks 0.6.9 has been changed after my problem and on my suggestion. Yet I not understood the reason my request has been closed. Please, can you can something to reopen my post? I not know someone with knowledges on libraries of the MiKTeX team. Best regards from Sicily.

**Comment by SSebastiano** [comment]
@edocevoli Good morning. Please can you reopen my request. The problem persist yet. Regards.

**Comment by SSebastiano** [comment]
@stloeffler Dear Stefan,

Regarding one of my previous questions, with the release of version 6.10 and the updated libraries, I can inform you that the bad rendering I had before is now gone.

What still remains is that, when I move the mouse, some traces occasionally remain visible. Could you tell me what I should do? Many users, including myself, have noticed this problem.

I am not sure whether some old TeXworks files are still present, and I don’t know what I could safely remove without causing damage. I am attaching a picture so that you can better understand my question.

What do you suggest?

<img width="861" height="698" alt="Image" src="https://github.com/user-attachments/assets/3a6f583a-4eba-490d-ae90-f7c6c0658539" />
Regards, Sebastiano

**Comment by stloeffler** [comment]
I'm happy that the bad rendering issue is resolved.
Regarding the artifact lines, I recently made a change to the code that should hopefully solve that issue as well. Could you try the latest development snapshot (see the [TeXworks homepage](https://tug.org/texworks/#Online_resources)) and check if the problem has been resolved?
